src/main/scala/io/iohk/ethereum/blockchain/sync/fast/FastSyncBranchResolver.scala

```diff
@@ -0,0 +1,182 @@
+package io.iohk.ethereum.blockchain.sync.fast
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props, Scheduler}
+import io.iohk.ethereum.blockchain.sync.PeerRequestHandler.{RequestFailed, ResponseReceived}
+import io.iohk.ethereum.blockchain.sync.fast.FastSyncBranchResolver._
+import io.iohk.ethereum.blockchain.sync.{BlacklistSupport, PeerListSupport, PeerRequestHandler}
+import io.iohk.ethereum.db.storage.AppStateStorage
+import io.iohk.ethereum.domain.{BlockHeader, Blockchain}
+import io.iohk.ethereum.network.EtcPeerManagerActor.PeerInfo
+import io.iohk.ethereum.network.Peer
+import io.iohk.ethereum.network.p2p.messages.Codes
+import io.iohk.ethereum.network.p2p.messages.PV62.{BlockHeaders, GetBlockHeaders}
+import io.iohk.ethereum.utils.Config.SyncConfig
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+
+class FastSyncBranchResolver(
+    val fastSync: ActorRef,
+    val peerEventBus: ActorRef,
+    val etcPeerManager: ActorRef,
+    val blockchain: Blockchain,
+    val syncConfig: SyncConfig,
+    val appStateStorage: AppStateStorage,
+    implicit val scheduler: Scheduler
+) extends Actor
+    with ActorLogging
+    with PeerListSupport
+    with BlacklistSupport {
```

Comment:
this will be a little problematic. 

As you can see with current encoding `BlacklistSupport` is per actor basis, this can lead to situation:

- In main fast sync actor, our master peer which is best peer failed (let say, sent us bad header), and we blacklisted him. We start `FastSyncBranchResolver` to chose new best peer and resolve branch
-  In `FastSyncBranchResolver` we get best peer and we get the same peer, as here he is not blacklisted
- we return to main fast sync actor and start sync with the same master.

Probably our blacklist, should be global for whole syncing component. to avoid it.